From 9123a9b0c0c03d4ce6ad5757fdfbc5d7c29b0ab1 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen.kooi@oss.qualcomm.com>
Date: Thu, 19 Jun 2025 16:06:14 +0200
Subject: [PATCH] dracut.sh: try $STRIP for $strip_cmd first

When using dracut in a cross enviroment, like OpenEmbedded, the host
provided strip (or eu-strip) won't work, so try using the $STRIP for the
shell environment first before falling back to path based lookups.

Signed-off-by: Koen Kooi <koen.kooi@oss.qualcomm.com>
Upstream-Status: Pending
---
 dracut.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dracut.sh b/dracut.sh
index 11da5c25..8facdb34 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -2269,7 +2269,8 @@ done
 if [[ $do_strip == yes ]]; then
     # Prefer strip from elfutils for package size
     declare strip_cmd
-    strip_cmd=$(command -v eu-strip)
+    strip_cmd="${STRIP}"
+    [ -z "$strip_cmd" ] && $(command -v eu-strip)
     [ -z "$strip_cmd" ] && strip_cmd="strip"
 
     for p in "$strip_cmd" xargs find; do
-- 
2.34.1

